src/H5Dchunk.c

```diff
@@ -59,6 +59,7 @@
 #include "H5Iprivate.h"  /* IDs                      */
 #include "H5MMprivate.h" /* Memory management            */
 #include "H5MFprivate.h" /* File memory management               */
+#include "H5PBprivate.h" /* Page Buffer	                         */
```

Comment:
There seems to be a lot of duplicated code between the read & write operations (more so now), could some of it be extracted as common code?

Reply:
Do you mean the array allocation/free code?

Reply:
No, I just put the comment at the top of the file, since it applied to all of your changes.

When reviewing this code, there's large chunks of common code that would benefit from refactoring out into common routines, which would reduce the size of the "main" routines, etc.

Reply:
Hmm, that and the file driver code is the only newly common code I can find

Reply:
I tried to do this (pulling out common code between the selection and non-selection branches of H5D__chunk_write) and it became a mess with the routine needing at least 15 parameters. Another option would be a macro that just uses the variable names as is (not as parameters), though that's ugly as well since changing the variable names breaks the macro. Thoughts?